lib/usage.ts

```diff
@@ -20,16 +20,22 @@ export function usage(yargs: YargsInstance, shim: PlatformShim) {
   self.failFn = function failFn(f) {
     fails.push(f);
   };
-  let failMessage: string | undefined | null = null;
+  let failMessage: string | nil = null;
+  let globalFailMessage: string | nil = null;
   let showHelpOnFail = true;
   self.showHelpOnFail = function showHelpOnFailFn(
     arg1: boolean | string = true,
     arg2?: string
   ) {
-    function parseFunctionArgs(): [boolean, string?] {
-      return typeof arg1 === 'string' ? [true, arg1] : [arg1, arg2];
+    const [enabled, message] =
+      typeof arg1 === 'string' ? [true, arg1] : [arg1, arg2];
+
+    // If global context, set globalFailMessage
+    // Addresses: https://github.com/yargs/yargs/issues/2085
+    if (yargs.getInternalMethods().getContext().commands.length === 0) {
```

Comment:
Is there an easier way to figure out that this is in the global context? (I can move this logic to a helper function and call it here)

Reply:
I like the idea of pulling this into a private helper method, then if we can think of a better way to detect we're in the global context we can swap out the logic.